src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -135,10 +146,15 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
   // Overridden to use array copying for efficiency where possible.
   override def addAll(elems: IterableOnce[A]): this.type = {
     elems match {
-      case elems: ArrayBuffer[_] =>
-        ensureSize(length + elems.length)
-        Array.copy(elems.array, 0, array, length, elems.length)
-        size0 = length + elems.length
+      case elems: collection.Iterable[A] =>
```

Comment:
Do we want to match `IndexdSeq` only, since we call `elems.size` below?

Reply:
I'm just copying what `insertAll` already does.

I think the rationale is that it's cheaper to iterate over the collection twice than to allocate a whole bunch of (increasingly large) arrays repeatedly.

side note: both this method and `insertAll` probably ought to have a case handling `IterableOnce` where `knownSize >= 0` as well

Reply:
> I think the rationale is that it's cheaper to iterate over the collection twice than to allocate a whole bunch of (increasingly large) arrays repeatedly.

the benchmarks do not appear to agree with this

Reply:
I ended up checking `knownSize`, which is slightly broader than `IndexedSeq`, but still cheap